proxy/http/Makefile.am

```
@@ -158,7 +160,8 @@ test_HttpTransact_LDADD = \
 	@LIBZ@ \
 	@LIBLZMA@ \
 	@OPENSSL_LIBS@ \
-	-lm
+	-lm \
+	@YAMLCPP_LIBS@
```

```suggestion
 	@YAMLCPP_LIBS@ \
	-lm
```

BEst of keep `-lm` last since order matters :)

---

I'm just a vessel for the PR. This is a backport of #9885 to 9.2.x. We should follow up with a another PR to make it last.